rewrite-java/src/main/java/org/openrewrite/java/JavaTemplate.java

```diff
@@ -417,7 +417,7 @@ public J visitMethodInvocation(J.MethodInvocation method, Integer integer) {
                     }
                     return m;
                 }
-                return maybeReplaceStatement(method, J.class, true, 0);
+                return maybeReplaceStatement(method, J.class, getCursor().firstEnclosing(J.VariableDeclarations.class) == null, 0);
```

Comment:
This isn't quite right, since the method invocation could be within an anonymous class declaration.
Consider if `bar()` were the subject of the replacement:
```
Object o = new Object {
    void foo() {
        if(bar()) {
        }
    }
};
```

In this case `bar()` should be treated like an expression and not like a statement.
I think this would be correct if you stopped looking for a `J.VariableDeclarations` on the cursor stack when encountering a class declaration or lamdba.